tracex_project/extraction/views.py

```diff
@@ -114,25 +105,37 @@ def get_context_data(self, **kwargs):
         """Prepare the data for the result page."""
         context = super().get_context_data(**kwargs)
         orchestrator = Orchestrator.get_instance()
-        event_types = orchestrator.get_configuration().event_types
-        filter_dict = {
-            "concept:name": event_types,
-            "attribute_location": orchestrator.get_configuration().locations,
-        }
-
-        output_path_xes = f"{str(utils.output_path / 'single_trace')}_event_type.xes"
-        single_trace_df = pm4py.read_xes(output_path_xes)
-        single_trace_df.rename(
-            columns={"time:timestamp": "start_timestamp"}, inplace=True
-        )
-
-        # 2. Sort and filter the single journey dataframe
-        single_trace_df = single_trace_df.sort_values(
-            by="start_timestamp", inplace=False
+        single_trace_df = orchestrator.get_data()
+        activity_key = orchestrator.get_configuration().activity_key
+
+        # 1. Set the filter dictionary based on the activity key
+        match (activity_key):
+            case "activity":
+                filter_dict = {
+                    "attribute_location": orchestrator.get_configuration().locations,
+                    "event_type": orchestrator.get_configuration().event_types,
+                }
+            case "event_type":
+                filter_dict = {
+                    "attribute_location": orchestrator.get_configuration().locations,
+                    "concept:name": orchestrator.get_configuration().event_types,
+                }
+            case "attribute_location":
+                filter_dict = {
+                    "concept:name": orchestrator.get_configuration().locations,
+                    "event_type": orchestrator.get_configuration().event_types,
+                }
+            case _:
+                filter_dict = {}
```

Comment:
Can this also be outsourced to the Orchestrator to a method like `get_activity_key`? For me it is a bit weird that we have this condition in the views even though it does not directly relate to something getting displayed

Reply:
Will be tackled in refactoring-2